### PR TITLE
Pipeline automation: create, run, delete

### PR DESCRIPTION
**Pull request by slickwarren**
moved pipeline delete to main test

ran flake8, added environment variables

added clarifying comments

moved test def to top of definitions, added delete test case

### CODE REVIEW

**Review by bmdepesa — APPROVED** [review]
LGTM

File: tests/validation/tests/v3_api/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    def fin():
+        # project delete doesn't remove all ns in test, removing manually
+        client = get_cluster_client_for_token(namespace["cluster"], USER_TOKEN)
+        for ns_x in client.list_namespace(projectId=namespace["project"].id):
+            client.delete(ns_x)
+        client.delete(namespace["project"])
+    request.addfinalizer(fin)
+
+
+def test_pipeline():
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_pipeline.py:36]: Can we have this test in the top - first, following by @pytest.fixture ?

File: tests/validation/tests/v3_api/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+
+    def fin():
+        # project delete doesn't remove all ns in test, removing manually
+        client = get_cluster_client_for_token(namespace["cluster"], USER_TOKEN)
+        for ns_x in client.list_namespace(projectId=namespace["project"].id):
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_pipeline.py:30]: Why do we need this? I see only one namespace get created under create_project_client()

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/test_pipeline.py (hunk 2 above)
Comment (slickwarren) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: The pipeline creates a ns for the example run with a random name.

Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: The pipeline creates it's own namespaces to which the artifacts are deployed.

**Comment by slickwarren** [comment]
closing in favor of #31596

**Pull request closed without merging**